Remove LibIRC logic from cmake.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31899 Revert "Move AutogradMeta and DeviceGuardImplInterface virtual methods out-of-line."
* #31162 Don't use RTLD_GLOBAL to load _C.
* #31161 Uniformly apply Windows logic in cpp_extensions everywhere
* #31236 Don't unconditionally compile runJITCPPTests
* #31176 Move AutogradMeta and DeviceGuardImplInterface virtual methods out-of-line.
* #31157 Add missing TORCH_CUDA_API annotation to throw_nccl_error
* #31155 Remove dead CAFFE2_LIBS variable
* **#31152 Remove LibIRC logic from cmake.**

Per apaszke: I can't find any reasonable references to libIRC online, so
I decided to remove this.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262582](https://our.internmc.facebook.com/intern/diff/D19262582)

ezyang committed:
Remove LibIRC logic from cmake.

Per apaszke: I can't find any reasonable references to libIRC online, so
I decided to remove this.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit d4571fa:
* **3/3** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d4571fa6591bb0a268924a0af8fcd2f3ba6750e2).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3914810) pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build (1/1)
**Step:** "pytorch android gradle build" ([details](https://dr.pytorch.org/build-details.html?build_id=72755339))
```
sys	0m0.183s
++ docker run --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -t -d -w /var/lib/jenkins 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-d4571fa6591bb0a268924a0af8fcd2f3ba6750e2-android-x86_32
+ export id_x86_32=16fef253e65e45b667f4e9142769a33b63e1afa4aedadd0d003cd682045d0b86
+ id_x86_32=16fef253e65e45b667f4e9142769a33b63e1afa4aedadd0d003cd682045d0b86
+ export 'COMMAND=((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ COMMAND='((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ echo '((echo' '"source' './workspace/env"' '&&' echo '"sudo' chown -R jenkins 'workspace")' '|' docker exec -u jenkins -i '"$id_x86_32"' 'bash)' '2>&1'
+ ts
+ unbuffer bash ./command.sh
+ docker pull 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-d4571fa6591bb0a268924a0af8fcd2f3ba6750e2-android-arm-v7a
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-d4571fa6591bb0a268924a0af8fcd2f3ba6750e2-android-arm-v7a not found
```
### 2 failures *not* recognized by patterns:
| Job | Step |
| --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>pytorch_parallelnative_linux_xenial_py3_6_gcc5_4_build</sup> | <sup>Build</sup> |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>pytorch_paralleltbb_linux_xenial_py3_6_gcc5_4_build</sup> | <sup>Build</sup> |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

ezyang committed:
Update on "Remove LibIRC logic from cmake."


Per apaszke: I can't find any reasonable references to libIRC online, so
I decided to remove this.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Remove LibIRC logic from cmake."


Per apaszke: I can't find any reasonable references to libIRC online, so
I decided to remove this.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Remove LibIRC logic from cmake."


Per apaszke: I can't find any reasonable references to libIRC online, so
I decided to remove this.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

seemethere reviewed (approved):
LGTM

The only real reference I could find for `libirc` is [here](https://software.intel.com/en-us/forums/intel-c-compiler/topic/799085) but it doesn't seem entirely relevant to this project.

ezyang committed:
Update on "Remove LibIRC logic from cmake."


Per apaszke: I can't find any reasonable references to libIRC online, so
I decided to remove this.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19262582](https://our.internmc.facebook.com/intern/diff/D19262582)

[ghstack-poisoned]
